src/Subscriptions/BroadcastManager.php

```diff
@@ -0,0 +1,202 @@
+     */
+    protected function get(string $name)
+    {
+        return $this->broadcasters[$name] ?? $this->resolve($name);
+    }
+
+    /**
+     * Get a broadcaster instance by name.
+     *
+     * @param string|null $name
+     *
+     * @return BroadcastsSubscriptions
+     */
+    public function broadcaster($name = null)
+    {
+        $name = $name ?: $this->getDefaultDriver();
+
+        return $this->broadcasters[$name] = $this->get($name);
+    }
+
+    /**
+     * Get the default broadcast driver name.
+     *
+     * @return string
+     */
+    public function getDefaultDriver()
+    {
+        return $this->app['config']['lighthouse.subscriptions.driver'];
+    }
+
+    /**
+     * Set the default broadcast driver name.
+     *
+     * @param string $name
+     */
+    public function setDefaultDriver($name)
+    {
+        $this->app['config']['lighthouse.subscriptions.driver'] = $name;
+    }
+
+    /**
+     * Get the subscription configuration.
+     *
+     * @return array
+     */
+    protected function getConfig()
+    {
+        return $this->app['config']['lighthouse.subscriptions'];
+    }
+
+    /**
+     * Register a custom driver creator Closure.
+     *
+     * @param string   $driver
+     * @param \Closure $callback
+     *
+     * @return self
+     */
+    public function extend($driver, \Closure $callback)
+    {
+        $this->customCreators[$driver] = $callback;
+
+        return $this;
+    }
+
+    /**
+     * Resolve the given broadcaster.
+     *
+     * @param string $name
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return BroadcastsSubscriptions
+     */
+    protected function resolve($name)
+    {
+        $config = $this->getConfig();
+
+        if (isset($this->customCreators[$name])) {
+            return $this->callCustomCreator($config);
+        }
+
+        $driverMethod = 'create'.ucfirst($name).'Driver';
+
+        if (! method_exists($this, $driverMethod)) {
+            throw new \InvalidArgumentException("Subscription driver [{$name}] is not supported.");
+        }
+
+        return $this->{$driverMethod}($config);
+    }
+
+    /**
+     * Call a custom driver creator.
+     *
+     * @param array $config
+     *
+     * @return mixed
+     */
+    protected function callCustomCreator(array $config)
+    {
+        return $this->customCreators[$config['driver']]($this->app, $config);
+    }
+
+    /**
+     * Create instance of pusher driver.
+     *
+     * @param array $config
+     *
+     * @return Broadcaster
+     */
+    protected function createPusherDriver(array $config)
+    {
+        $appKey = config('broadcasting.pusher.key');
+        $appSecret = config('broadcasting.pusher.secret');
+        $appId = config('broadcasting.pusher.app_id');
+        $appCluster = config('broadcasting.pusher.options.cluster');
+
+        $pusher = new \Pusher\Pusher($appKey, $appSecret, $appId, [
+            'cluster' => $appCluster,
```

Comment:
Just a quick suggestion: It would be useful to pass the whole pusher config options array. The use case is when a third party Pusher server is used (like laravel-websockets).

Reply:
Great suggestion makes total sense, thanks!!